### PR TITLE
feat(datasource/kurora): crossbell and optimism and avalanche support

### PR DESCRIPTION
**Pull request by kallydev**

### CODE REVIEW

File: deploy/config/indexer.yaml [hunk 1]
```diff
@@ -14,6 +14,9 @@ ethereumetl:
   password: password
   database: etl
 
+kurora:
```
Comment (polebug) [inline deploy/config/indexer.yaml:17]: Maybe it's better to put in rpc -> pregod_etl -> kurora?

Comment (kallydev) [reply deploy/config/indexer.yaml:17]: i think kurora is not part of etl 👀

Comment (kallydev) [reply deploy/config/indexer.yaml:17]: and etl will be replaced in the future

**Review by polebug — APPROVED** [review]

File: deploy/config/indexer.yaml (hunk 1 above)
Comment (kallydev) [reply deploy/config/indexer.yaml:17]: and it and etl are not rpc, rpc should only store jsonrpc endpoints of blockchain nodes

File: service/indexer/internal/datasource/kurora/datasource.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+		// The receipts are obtained here instead of the transactions,
+		// because the receipts do not require additional requests to get the status of the transactions.
+		// Only fetch transaction receipts where the `receipt.from` is `message.Address`.
+		receipts, err := d.kuroraClient.FetchEthereumReceipts(ctx, network, ethereumReceiptQuery)
+		if err != nil {
+			return nil, fmt.Errorf("ethereum transactions: %w", err)
+		}
+
+		if len(receipts) == 0 {
```
Comment (brucexc) [inline service/indexer/internal/datasource/kurora/datasource.go]: This length judgment can be removed because of the following 'range' iteration.

**Review by brucexc — APPROVED** [review]

**Review by HenryQW — APPROVED** [review]

**Pull request merged**